web/src/user_group_edit_members.ts

```diff
@@ -229,10 +263,17 @@ function add_new_members({pill_user_ids}) {
         });
     }
 
-    function invite_failure(xhr) {
+    function invite_failure(xhr?: JQuery.jqXHR): void {
         let message = "Failed to add user!";
-        if (xhr.responseJSON?.msg) {
-            message = xhr.responseJSON.msg;
+
+        const request_failure_response_schema = z
+            .object({
+                msg: z.string().min(1),
+            })
```

Comment:
The way we mark all JSON errors is with `"result": "error"`.

```suggestion
            .object({
                result: z.literal("error"),
                msg: z.string(),
            })
```